### PR TITLE
Non-uplink PDAs no longer can have TC inserted into them

### PR DESCRIPTION
**Pull request by Zonespace27**
## About the PR

PDAs without an uplink no longer can have TC inserted. Fixes https://github.com/space-wizards/space-station-14/issues/28960

## Why / Balance

Two reasons:
- Obviously, non-syndicate PDAs shouldn't take TC
- The ability to insert TC into PDAs only extends to regular PDAs and unlocked syndicate PDAs, allowing players to tell if a PDA is actually a locked uplink as well. This is obviously bad as well.

## Technical details

Listens for a `CurrencyInsertAttemptEvent` event on the `RingerComponent` instead of `RingerUplinkComponent`, and checks for the latter when the event fires. Should there not be a ringerUplink, then the TC insertion is cancelled.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- fix: Non-uplink PDAs can no longer have telecrystals inserted into them.

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
1] codewise, you need to add an event rather than handling it directly 
2] this enables people to metagame trivially by acquiring TC and then attempting to insert it into a PDA to check whether theres an uplink or not

File: Content.Server/Store/Systems/StoreSystem.cs [hunk 1]
```diff
@@ -87,6 +88,10 @@ private void OnAfterInteract(EntityUid uid, CurrencyComponent component, AfterIn
         if (!TryComp<StoreComponent>(args.Target, out var store))
             return;
 
+        // PDAs without uplinks shouldn't be allowed to have currency added
+        if (TryComp<RingerComponent>(args.Target, out var ringer) && !TryComp<RingerUplinkComponent>(args.Target, out var ringerUplink))
```
Comment (Zonespace27) [inline Content.Server/Store/Systems/StoreSystem.cs:92]: Like I said in the technical details header, I don't really like how this is done. While it _works_ this features PDA uplink code in the generic store code, which is a bit anti-OOP. However, I'm limited by the fact that I don't know the codebase well enough to do it in a better way. Suggestions or pointers for an improved way are appreciated.

**Comment by Zonespace27** [comment]
> 1] codewise, you need to add an event rather than handling it directly 

Will look into, thanks

> 2] this enables people to metagame trivially by acquiring TC and then attempting to insert it into a PDA to check whether theres an uplink or not

This PR explicitly fixes that.

**Comment by EmoGarbage404** [comment]
ngl i didnt even know we had that event i thought you were going to have to code it lol

**Comment by FairlySadPanda** [comment]
1 Go onto bridge
2 murder the fox
3 use scavenged TC to syndie check

**Comment by EmoGarbage404** [comment]
> 1 Go onto bridge 2 murder the fox 3 use scavenged TC to syndie check

from my understanding, you already can't insert TC into locked uplinks. The issue, which this solves, is that if a PDA can't lock (i.e. isn't an uplink) the check fails and you can insert it anyways.

the logic is kinda weird

**Pull request merged**